src/codegen/abi.js

```diff
@@ -59,18 +60,60 @@ module.exports = class AbiCodeGenerator {
             name = `param${index}`
           }
           paramsClass.addMethod(
-            tsCodegen.method(
+            input.get('type') === 'tuple' ?
+              tsCodegen.method(
+                `get ${name}`,
+                [],
+                tsCodegen.namedType(name).capitalize(),
+                `
+            return <${tsCodegen.namedType(name).capitalize()}>${typesCodegen.ethereumValueToAsc(
+                  `this._event.parameters[${index}].value`,
+                  input.get('type')
+                )}
+            `
+              ) :
+              tsCodegen.method(
+                  `get ${name}`,
+                  [],
+                   typesCodegen.ascTypeForEthereum(input.get('type')),
+                  `
+                return ${typesCodegen.ethereumValueToAsc(
+                  `this._event.parameters[${index}].value`,
+                  input.get('type')
+                )}
+                `
+              )
+          )
+        })
+
+        // Generate param getters differently for Tuple params
+        // Also generate a class for each Tuple param
+        let tupleKlasses = [];
```

Comment:
We can use "Classes" and "Class" in this part of the code since there's a prefix in the name, so `tupleClasses`, `tupleClass` like with `paramClass`.